types/properties-reader/index.d.ts

```diff
@@ -0,0 +1,26 @@
+// Type definitions for properties-reader 0.0
+// Project: https://github.com/steveukx/properties
+// Definitions by: Zlatko Andonovski <https://github.com/Goldsmith42/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare namespace PropertiesReader {
+    type Value = string|number|boolean;
+
+    interface Reader {
+        get(propertyName: string): Value|null;
+        getRaw(propertyName: string): string|null;
+        path(): Object;
+        append(path: string): Reader;
+        read(properties: string): Reader;
+        set(propertyName: string, value: Value): Reader;
+        length: number;
+        each(iterator: (key: string, value: Value) => void, scope?: Object): Reader;
```

Comment:
If you want support for a this-parameter you could add an overload:
```ts
each(iterator: (key: string, value: Value) => void): Reader;
each(iterator: (this: T, key: string, value: Value) => void, scope?: T): Reader;
```
Be sure to test.